Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs

```diff
@@ -0,0 +1,241 @@
+            if (!Directory.Exists(serverFolderPath)) {
+                MessageBox.Show(Strings.LanguageClientNodejsServerNotFound, Strings.ProductTitle);
+                return null;
+            }
+
+            var serverFilePath = Path.Combine(serverFolderPath, @"server\server.bundle.js");
+            if (!File.Exists(serverFilePath)) {
+                MessageBox.Show(Strings.LanguageClientNodejsServerNotFound, Strings.ProductTitle);
+                return null;
+            }
+
+            await Task.Yield();
+
+            var info = new ProcessStartInfo {
+                FileName = nodePath,
+                WorkingDirectory = serverFolderPath,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                Arguments = serverFilePath,
+            };
+
+            var process = new Process {
+                StartInfo = info
+            };
+
+            if (process.Start()) {
+                return new Connection(process.StandardOutput.BaseStream, process.StandardInput.BaseStream);
+            }
+
+            return null;
+        }
+
+        public override object CreateInitializationOptions(string interpreterPath, string interpreterVersion, string rootPath, IEnumerable<string> searchPaths) {
+            return new PythonInitializationOptions {
+                analysis = null,
+                pythonPath = interpreterPath,
+                disableLanguageServices = false,
+                openFilesOnly = false,
+                venvPath = null,
+                useLibraryCodeForTypes = true,
+            };
+        }
+
+        private static string GetServerLocation() {
+            // At some point, we'll bundle it and return the install location
+            // but for now, to allow for testing, location is retrieved via
+            // environment variable.
+            var folderPath = Environment.GetEnvironmentVariable("PTVS_NODE_SERVER_LOCATION");
+            if (Directory.Exists(folderPath)) {
+                return folderPath;
+            }
+
+            return null;
+        }
+
+        private static bool IsEnvVarEnabled(string variable) {
+            var val = Environment.GetEnvironmentVariable(variable);
+            if (string.IsNullOrEmpty(val)) {
+                return false;
+            }
+
+            return val != "0" && string.Compare(val, "false", StringComparison.OrdinalIgnoreCase) != 0;
+        }
+
+        private string GetNodeExecutablePath() {
+            string filePath;
+
+            // Prefer the node executable from the server folder, if there is one
+            var serverFolderPath = GetServerLocation();
+            if (Directory.Exists(serverFolderPath)) {
+                filePath = Path.Combine(serverFolderPath, "node.exe");
+                if (File.Exists(filePath)) {
+                    return filePath;
+                }
```

Comment:
do we care about logging errors?

Reply:
right now they'll get a message box if either node or the server is missing.